src/engine/strat_engine/cleanup.rs

```diff
@@ -29,3 +31,20 @@ pub fn wipe_blockdevs(mut blockdevs: Vec<BlockDev>) -> EngineResult<()> {
         Err(EngineError::Engine(ErrorEnum::Error, err_msg))
     }
 }
+
+/// Teardown pools.
+pub fn teardown_pools(mut pools: Vec<StratPool>) -> EngineResult<()> {
+    let mut untorndown_pools = Vec::new();
+    for pool in pools.drain(..) {
+        let pool_uuid = *pool.uuid();
+        pool.teardown()
+            .unwrap_or_else(|_| untorndown_pools.push(pool_uuid));
+    }
+    if untorndown_pools.is_empty() {
+        Ok(())
+    } else {
```

Comment:
It matches the one above, so I'ld like to leave it like that. The hope is that this stuff can be generalized somehow, but it is not quite obvious how, yet.